### PR TITLE
#42 preliminary lambda + scheduler for data_upload

### PR DESCRIPTION
**Pull request by S1mpySloth**
#42  created a preliminary lambda + weekly scheduler tf script. Has not been `terraform apply` 'd yet.

### CODE REVIEW

File: terraform/data_upload_lambda.tf [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    ]
+  })
+}
+
+# Lambda Function using Docker Image
+resource "aws_lambda_function" "c14_lambda_function" {
+  function_name    = "c14-earthquake-monitor-etl-lambda"
+  role             = aws_iam_role.c14_lambda_execution_role.arn
+  package_type     = "Image"
+  image_uri        = "${var.ETL_ECR_URI}:latest"
```
Comment (Jakub-Poskrop) [inline terraform/data_upload_lambda.tf:55]: You might have to change this as the latest ECR image is usually an image index that doesn't work with lambdas.

File: terraform/data_upload_lambda.tf [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        Effect   = "Allow"
+        Resource = aws_lambda_function.c14_lambda_function.arn
+      }
+    ]
+  })
+}
+
+# EventBridge Schedule
+resource "aws_scheduler_schedule" "c14_scheduler_schedule" {
+  name                         = "c14-earthquake-monitor-etl-schedule"
```
Comment (Jakub-Poskrop) [inline terraform/data_upload_lambda.tf:110]: Maybe change the name to something more descriptive like `c14-earthquake-report-schedule`?

File: terraform/data_upload_lambda.tf [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        Effect   = "Allow"
+        Resource = "arn:aws:s3:::${var.S3_BUCKET}/*"
+      }
+    ]
+  })
+}
+
+# Lambda Function using Docker Image
+resource "aws_lambda_function" "c14_lambda_function" {
+  function_name    = "c14-earthquake-monitor-etl-lambda"
```
Comment (Jakub-Poskrop) [inline terraform/data_upload_lambda.tf:52]: Maybe change the name to something more descriptive like `c14-earthquake-report-lambda`?

File: terraform/data_upload_lambda.tf (hunk 1 above)
Comment (Jakub-Poskrop) [reply terraform/data_upload_lambda.tf:55]: Ok all good

File: terraform/data_upload_lambda.tf (hunk 3 above)
Comment (S1mpySloth) [reply terraform/data_upload_lambda.tf]: Done!

**Review by Jakub-Poskrop — APPROVED** [review]
Looks good

**Review by Fahi28 — APPROVED** [review]
nice stuff

**Pull request merged**